src/Common/Crypto/X509Certificate.h

```diff
@@ -0,0 +1,333 @@
+#pragma once
+
+#include <Common/Exception.h>
+
+#include <boost/container/flat_set.hpp>
+
+#include "config.h"
+
+#if USE_SSL
+#    include <openssl/bio.h>
+#    include <openssl/crypto.h>
+#    include <openssl/err.h>
+#    include <openssl/pem.h>
+#    include <openssl/x509.h>
+#    include <openssl/x509v3.h>
+#    include <Common/OpenSSLHelpers.h>
+#endif
+
+#include <memory>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int OPENSSL_ERROR;
+    extern const int BAD_ARGUMENTS;
+}
+
+
+#if USE_SSL
+class X509Certificate
+{
+public:
+    using List = std::vector<X509Certificate>;
+    constexpr static size_t NAME_BUFFER_SIZE = 256;
```

Comment:
Sure about the size? We were previously using much larger buffers https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/System/StorageSystemCertificates.cpp#L77

EDIT: we pass in the buffer size whereever the buffer is used below. Should be good then.